### PR TITLE
Simple post message and Change name from slack-notification to slack-notifier

### PR DESCRIPTION
**Pull request by koudaiii**
## WHY

Simple post message to slack.

image

### CODE REVIEW

**Comment by dtan4** [comment]
Please define `bin/slack-notification` target.

``` bash
$ make
make: *** No rule to make target `bin/slack-notification'.  Stop.
```

File: main.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	"os"
+	"strings"
+)
+
+func main() {
+	// input send text
+	args := strings.Join(os.Args[1:], " ")
+	if args == "" {
+		fmt.Println("no text")
+		os.Exit(0)
```
Comment (dtan4) [inline main.go:33]: This should show command line usage and exit with 1.

```
usage: slack-notification <TEXT>
```

File: main.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	if err != nil {
+		fmt.Println("%v", err)
+		os.Exit(1)
+	}
+
+	//Form JSON payload to send to Slack
+	json := `{"text": "` + args + `"}`
+	//Post JSON payload to the Webhook URL
+	client := http.Client{}
+	req, err := http.NewRequest("POST", config.SlackWebhookURL, bytes.NewBufferString(json))
```
Comment (dtan4) [inline main.go]: Please handle this error right here.

File: main.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	req.Header.Set("Content-Type", "application/json")
+	_, err = client.Do(req)
+	if err != nil {
+		fmt.Println("Unable to reach the server.")
+	}
+	defer req.Body.Close()
```
Comment (dtan4) [inline main.go]: If this function does nothing after here, `defer` is unnecessary.

**Review by dtan4 — CHANGES_REQUESTED** [review]

File: main.go (hunk 2 above)
Comment (koudaiii) [reply main.go]: Add parse error handling.

```
 666    func NewRequest(method, urlStr string, body io.Reader) (*Request, error) {
   667      if method == "" {
   668          // We document that "" means "GET" for Request.Method, and people have
   669          // relied on that from NewRequest, so keep that working.
   670          // We still enforce validMethod for non-empty methods.
   671          method = "GET"
   672      }
   673      if !validMethod(method) {
   674          return nil, fmt.Errorf("net/http: invalid method %q", method)
   675      }
   676      u, err := url.Parse(urlStr)
   677      if err != nil {
   678          return nil, err
   679      }
```

File: main.go (hunk 3 above)
Comment (koudaiii) [reply main.go]: Delete `defer`.

**Comment by dtan4** [comment]
You may have to run `glide update` to fill dependencies.

``` bash
$ make
glide install
[WARN]  The name listed in the config file (github.com/koudaiii/slack-notifier) does not match the current location (github.com/wantedly/slack-notifier)
[WARN]  Lock file may be out of date. Hash check of YAML failed. You may need to run 'update'
```

``` diff
diff --git a/glide.lock b/glide.lock
index 46572fc..ff790c1 100644
--- a/glide.lock
+++ b/glide.lock
@@ -1,4 +1,8 @@
-hash: 99cd0a7bdab40e1fe748a40176e545dbf2dbf2efb1ccacd6adaf7a84b05581ae
-updated: 2016-09-25T23:50:06.896287729+09:00
-imports: []
-devImports: []
+hash: ebc3d350539b7a3189d8ce511aa53c2729338ac39e8f11a4e2520e2371a722f5
+updated: 2016-09-27T10:05:32.483310586+09:00
+imports:
+- name: github.com/kelseyhightower/envconfig
+  version: cea086319492c3940683ea5e122aa5de70a33923
+- name: github.com/pkg/errors
+  version: 01fa4104b9c248c8945d14d9f128454d5b28d595
+testImports: []
```

File: main.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+		os.Exit(1)
+	}
+
+	//Form JSON payload to send to Slack
+	json := `{"text": "` + args + `"}`
+	//Post JSON payload to the Webhook URL
+	client := http.Client{}
+	req, err := http.NewRequest("POST", config.SlackWebhookURL, bytes.NewBufferString(json))
+	if err != nil {
+		fmt.Println("Unable to parse slack webhook url.")
```
Comment (dtan4) [inline main.go]: `os.Exit(1)` should exist here, shouldn't it?

**Review by dtan4 — CHANGES_REQUESTED** [review]

**Comment by dtan4** [comment]
All error messages should be printed in stderr.

``` go
fmt.Fprintln(os.Stderr, err)
```

**Comment by dtan4** [comment]
It seems no dependency is listed in `glide.yaml`. Don't you forget to do these?

``` bash
$ glide get github.com/kelseyhightower/envconfig
$ glide get github.com/pkg/errors
```

File: config.go [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+func LoadConfig() (*SlackConfiguration, error) {
+	var config SlackConfiguration
+	err := envconfig.Process(ConfigPrefix, &config)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "Failed to load config from envs.")
+	}
+
+	return &config, nil
+}
```
Comment (dtan4) [inline config.go]: Does it really need? If there is no plan to add more configuration, using `os.Getenv("SLACK_WEBHOOK_URL")` is simple and readable.

**Review by dtan4 — CHANGES_REQUESTED** [review]

File: config.go (hunk 5 above)
Comment (koudaiii) [reply config.go]: slack-notifier is very simple application.
ok, I will change `LoadConfig`.

**Comment by koudaiii** [comment]
please review 🙇

**Review by dtan4 — APPROVED** [review]

**Comment by dtan4** [comment]
LGTM 👍

**Pull request merged**